src/software/simulation/simulator.cpp

```diff
@@ -11,63 +11,88 @@ extern "C"
 #include "firmware/app/world/firmware_world.h"
 }
 
-Simulator::Simulator(const World& world)
-    : physics_world(world.field()), friendly_goalie_id(world.friendlyTeam().getGoalieID())
+Simulator::Simulator(const Field& field) : physics_world(field) {}
+
+void Simulator::setBallState(const BallState& ball_state)
 {
-    physics_world.setBallState(world.ball().currentState().ballState());
-    // Note: The simulator currently makes the invariant that friendly robots
-    // are yellow robots, and enemies are blue. This will be fixed in
-    // https://github.com/UBC-Thunderbots/Software/issues/1325
-    std::vector<RobotStateWithId> yellow_robots;
-    for (const auto& robot : world.friendlyTeam().getAllRobots())
-    {
-        RobotStateWithId state{.id          = robot.id(),
-                               .robot_state = robot.currentState().robotState()};
-        yellow_robots.emplace_back(state);
-    }
-    physics_world.addYellowRobots(yellow_robots);
-    std::vector<RobotStateWithId> blue_robots;
-    for (const auto& robot : world.enemyTeam().getAllRobots())
+    physics_world.setBallState(ball_state);
+    simulator_ball = std::make_shared<SimulatorBall>(physics_world.getPhysicsBall());
+}
+
+void Simulator::removeBall()
+{
+    simulator_ball.reset();
+    physics_world.removeBall();
+}
+
+void Simulator::addYellowRobots(const std::vector<RobotStateWithId>& robots)
+{
+    physics_world.addYellowRobots(robots);
+    for (auto physics_robot : physics_world.getYellowPhysicsRobots())
     {
-        RobotStateWithId state{.id          = robot.id(),
-                               .robot_state = robot.currentState().robotState()};
-        blue_robots.emplace_back(state);
+        auto simulator_robot = std::make_shared<SimulatorRobot>(physics_robot);
+        auto firmware_robot  = SimulatorRobotSingleton::createFirmwareRobot();
+        auto firmware_ball   = SimulatorBallSingleton::createFirmwareBall();
+        // Release ownership of the pointers so the firmware_world can take ownership
```

Comment:
So the `FirmwareWorld` won't actually "take ownership" of these pointers, they won't be de-allocated along with the `FirmwareWorld`.

Reply:
true, removed the comment since it's not that helpful anyway